code/game/objects/items/miscellaneous.dm

```diff
@@ -101,6 +101,51 @@
 	new /obj/item/toy/crayon/spraycan(src)
 	new /obj/item/clothing/shoes/sandal(src)
 
+/obj/item/choice_beacon/augments
+	name = "augment beacon"
+	desc = "Summons augmentations. Can be used 3 times!"
+	var/uses = 3
+
+/obj/item/choice_beacon/augments/generate_display_names()
+	var/static/list/augment_list
+	if(!augment_list)
+		augment_list = list()
+		var/list/templist = list(
+		/obj/item/organ/cyberimp/brain/anti_drop,
+		/obj/item/organ/cyberimp/arm/toolset,
+		/obj/item/organ/cyberimp/arm/surgery,
+		/obj/item/organ/cyberimp/chest/thrusters,
+		/obj/item/organ/lungs/cybernetic/upgraded,
+		/obj/item/organ/liver/cybernetic/upgraded) //cyberimplants range from a nice bonus to fucking broken bullshit so no subtypesof
+		for(var/V in templist)
+			var/atom/A = V
+			augment_list[initial(A.name)] = A
+	return augment_list
+
+/obj/item/choice_beacon/augments/generate_options(mob/living/M)
+	var/list/display_names = generate_display_names()
+	if(!display_names.len)
+		return
+	var/choice = input(M,"Which item would you like to order?","Select an Item") as null|anything in display_names
+	if(!choice || !M.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
+		return
+
+	spawn_option(display_names[choice],M)
+	uses--
+	if(!uses)
+		qdel(src)
+	else
+		to_chat(M, "<span class='notice'>[uses] use[uses > 1 ? "s" : ""] remaining on the [src].</span>")
+
+/obj/item/choice_beacon/augments/spawn_option(obj/choice,mob/living/M)
+	new choice(get_turf(M))
+	var/msg = "<span class=danger>Something warps to your feet!</span>"
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(istype(H.ears, /obj/item/radio/headset))
```

Comment:
Seems bit weird to lock only message to having the headset.

Reply:
I just thought it was a nice bit of fluff, but actually this might make sense for the warp to be a visible message and this to only come in if you have a headset